decode.go

```diff
@@ -55,6 +55,10 @@ type UnmarshalOptions struct {
 		protoregistry.ExtensionTypeResolver
 	}
 
+	// If AllowPartial is set, input for messages that will result in missing
+	// required fields will not return an error.
+	AllowPartial bool
```

Comment:
When I added an test editions file, and used features to toggle use of groups and required fields, I discovered that required fields weren't correctly enforced. This comment and field name come directly from corresponding fields in `proto.UnmarshalOptions` and `protojson.UnmarshalOptions`. So, by default, this will complain if the message has required fields that the YAML contents failed to set.